cli/determined_cli/shell.py

```diff
@@ -70,13 +70,14 @@ def _open_shell(shell: Command, additional_opts: str) -> None:
         if host == LOOPBACK_ADDRESS:
             host = "localhost"
 
+        proxy = "-o ProxyCommand='python -m determined_cli.tunnel {} %h:%p'".format(master)
         username = shell.agent_user_group["user"] or "root"
 
-        os.system(
-            "ssh -o StrictHostKeyChecking=no -tt -o IdentitiesOnly=yes -i {} -p {} {} {}@{}".format(
-                fp.name, port, additional_opts, username, host
-            )
-        )
+        cmd = "ssh {} -o StrictHostKeyChecking=no -tt -o IdentitiesOnly=yes -i {} -p {} {} {}@{}"
```

Comment:
Not in this PR, but this is totally one of the follow-up PRs I mentioned to you.  Check out the latest commit on my `shell-cli` branch.

(I'd also like to change the shell CLI just just pass all extra arguments to ssh, rather than use `shlex` and the whole double-shell-escaping paradigm)